sdk/main/include/Channel.h

```diff
@@ -67,8 +60,19 @@ class Channel
    *
    * @param url The URL and port of the gRPC service with which this channel
    *            should communicate.
+   * @param requireTLS initializes the channel with TLS if true, otherwise initializes an unsecured channel
+   * @param nodeCertHash the sha-384 hash of the node certificate chain. only relevant if requireTLS is true
+   *
+   * @return true if channel was successfully initialized, otherwise false
    */
-  void initChannel(const std::string& url);
+  bool initializeChannel(const std::string& url, bool requireTLS, const std::string& nodeCertHash = "");
```

Comment:
Could this function just check if `nodeCertHash` is empty? Or is `requireTLS` required for something else that I'm missing?

Reply:
I made separate functions to get rid of the yuckiness